---
title: "F~st~ Summary"
author: "Will Pitchers"
date: "9 November 2015"
output: html_document
---

```{r set up env, results='hide', echo=FALSE, message=FALSE}
rm( list=ls( ))

setwd( "/Volumes/Old_HD/Dropbox (MSU Efish Lab)/WILL/genome_assembly/garrulous-turtle/" )

require( dplyr )
require( ggplot2 )
require( ggExtra )
require( reshape )

APA_BAM <- read.table( "Fsts/Fst_APA_vs_BAM.windowed100kb.step50kb.weir.fst", header=TRUE )
APA_BAVA <- read.table( "Fsts/Fst_APA_vs_BAVA.windowed100kb.step50kb.weir.fst", header=TRUE )
APA_COB <- read.table( "Fsts/Fst_APA_vs_COB.windowed100kb.step50kb.weir.fst", header=TRUE )
APA_IVI <- read.table( "Fsts/Fst_APA_vs_IVI.windowed100kb.step50kb.weir.fst", header=TRUE )
APA_MOV <- read.table( "Fsts/Fst_APA_vs_MOV.windowed100kb.step50kb.weir.fst", header=TRUE )
BAM_BAVA <- read.table( "Fsts/Fst_BAM_vs_BAVA.windowed100kb.step50kb.weir.fst", header=TRUE )
BAM_COB <- read.table( "Fsts/Fst_BAM_vs_COB.windowed100kb.step50kb.weir.fst", header=TRUE )
BAM_IVI <- read.table( "Fsts/Fst_BAM_vs_IVI.windowed100kb.step50kb.weir.fst", header=TRUE )
BAM_MOV <- read.table( "Fsts/Fst_BAM_vs_MOV.windowed100kb.step50kb.weir.fst", header=TRUE )
BAVA_COB <- read.table( "Fsts/Fst_BAVA_vs_COB.windowed100kb.step50kb.weir.fst", header=TRUE )
BAVA_IVI <- read.table( "Fsts/Fst_BAVA_vs_IVI.windowed100kb.step50kb.weir.fst", header=TRUE )
BAVA_MOV <- read.table( "Fsts/Fst_BAVA_vs_MOV.windowed100kb.step50kb.weir.fst", header=TRUE )
COB_IVI <- read.table( "Fsts/Fst_COB_vs_IVI.windowed100kb.step50kb.weir.fst", header=TRUE )
COB_MOV <- read.table( "Fsts/Fst_COB_vs_MOV.windowed100kb.step50kb.weir.fst", header=TRUE )
IVI_MOV <- read.table( "Fsts/Fst_IVI_vs_MOV.windowed100kb.step50kb.weir.fst", header=TRUE )

```

The first section of these results draws from the results files generated with a 100kb window, stepped at 50kb intervals.

```{r data munging, results='hide', echo=FALSE, message=FALSE}

sum_APA_BAM <- group_by( APA_BAM, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_APA_BAVA <- group_by( APA_BAVA, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_APA_COB <- group_by( APA_COB, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_APA_IVI <- group_by( APA_IVI, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_APA_MOV <- group_by( APA_MOV, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_BAM_BAVA <- group_by( BAM_BAVA, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_BAM_COB <- group_by( BAM_COB, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_BAM_IVI <- group_by( BAM_IVI, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_BAM_MOV <- group_by( BAM_MOV, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_BAVA_COB <- group_by( BAVA_COB, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_BAVA_IVI <- group_by( BAVA_IVI, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_BAVA_MOV <- group_by( BAVA_MOV, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )
sum_COB_IVI <- group_by( COB_IVI, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_COB_MOV <- group_by( COB_MOV, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_IVI_MOV <- group_by( IVI_MOV, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )


mydataframes <- list( sum_APA_BAM=sum_APA_BAM, sum_APA_BAVA=sum_APA_BAVA, sum_APA_COB=sum_APA_COB, sum_APA_IVI=sum_APA_IVI, sum_APA_MOV=sum_APA_MOV, sum_BAM_BAVA=sum_BAM_BAVA, sum_BAM_COB=sum_BAM_COB, sum_BAM_IVI=sum_BAM_IVI, sum_BAM_MOV=sum_BAM_MOV, sum_BAVA_COB=sum_BAVA_COB, sum_BAVA_IVI=sum_BAVA_IVI, sum_BAVA_MOV=sum_BAVA_MOV, sum_COB_IVI=sum_COB_IVI, sum_COB_MOV=sum_COB_MOV, sum_IVI_MOV=sum_IVI_MOV )

cor_fst <- rep(NA, 15)

for (i in 1:15) {
  cor_fst[i] <- cor( mydataframes[[i]][2], mydataframes[[i]][3] )
  new_names <- paste( names( mydataframes[i] ), names( mydataframes[[i]] ), sep='_', coll='' )
  names( mydataframes[[i]] ) <- c( "SCAF", new_names[ 2:4 ] )
}

mean( cor_fst )

fsts <- mydataframes$sum_APA_BAM

for (i in 2:15) {
fsts <- merge( fsts, mydataframes[[i]], by.x="SCAF", by.y="SCAF", all.x=TRUE )
  }

```

Firstly, the VCFtools F~st~ calculator emits both 'raw' and 'adjusted' F~st~ values, so let's take a look at how closely these two measures predict one another... 

The plot below shows the relationship of the two F~st~ estimates across all scaffolds and all 15 comparisons (the red line is 1:1). The R^2^ for the correlation here is `r round(mean( cor_fst ),2)`, suggesting that it *probably* doesn't matter a great deal which statistic we use -- though the adjusted F~st~ is generally somewhat higher.

```{r fst vs adj fst plot, message=FALSE, echo=FALSE, results='hide', warning=FALSE, fig.width=7, fig.height=7}

names(fsts)

ggplot( fsts, aes( sum_APA_BAM_w_fst_, sum_APA_BAM_m_fst_ ) ) + geom_point( col=rgb( 0.1, 0.1, 0.1, 0.1 ) ) + 
        geom_point( aes( sum_APA_BAVA_w_fst_, sum_APA_BAVA_m_fst_ ), col=rgb( 0.1, 0.1, 0.1, 0.1 ) ) +
        geom_point( aes( sum_APA_COB_w_fst_, sum_APA_COB_m_fst_ ), col=rgb( 0.1, 0.1, 0.1, 0.1 ) ) +
        geom_point( aes( sum_APA_IVI_w_fst_, sum_APA_IVI_m_fst_ ), col=rgb( 0.1, 0.1, 0.1, 0.1 ) ) +
        geom_point( aes( sum_APA_MOV_w_fst_, sum_APA_MOV_m_fst_ ), col=rgb( 0.1, 0.1, 0.1, 0.1 ) ) +
        geom_point( aes( sum_BAM_BAVA_w_fst_, sum_BAM_BAVA_m_fst_ ), col=rgb( 0.1, 0.1, 0.1, 0.1 ) ) +
        geom_point( aes( sum_BAM_COB_w_fst_, sum_BAM_COB_m_fst_ ), col=rgb( 0.1, 0.1, 0.1, 0.1 ) ) +
        geom_point( aes( sum_BAM_IVI_w_fst_, sum_BAM_IVI_m_fst_ ), col=rgb( 0.1, 0.1, 0.1, 0.1 ) ) +
        geom_point( aes( sum_BAM_MOV_w_fst_, sum_BAM_MOV_m_fst_ ), col=rgb( 0.1, 0.1, 0.1, 0.1 ) ) +
        geom_point( aes( sum_BAVA_COB_w_fst_, sum_BAVA_COB_m_fst_ ), col=rgb( 0.1, 0.1, 0.1, 0.1 ) ) +
        geom_point( aes( sum_BAVA_IVI_w_fst_, sum_BAVA_IVI_m_fst_ ), col=rgb( 0.1, 0.1, 0.1, 0.1 ) ) +
        geom_point( aes( sum_BAVA_MOV_w_fst_, sum_BAVA_MOV_m_fst_ ), col=rgb( 0.1, 0.1, 0.1, 0.1 ) ) +
        geom_point( aes( sum_COB_IVI_w_fst_, sum_COB_IVI_m_fst_ ), col=rgb( 0.1, 0.1, 0.1, 0.1 ) ) +
        geom_point( aes( sum_COB_MOV_w_fst_, sum_COB_MOV_m_fst_ ), col=rgb( 0.1, 0.1, 0.1, 0.1 ) ) +
        geom_point( aes( sum_IVI_MOV_w_fst_, sum_COB_IVI_m_fst_ ), col=rgb( 0.1, 0.1, 0.1, 0.1 ) ) +
    xlab( "adjusted Fst" ) + ylab( "Fst" ) + geom_abline( intercept=0, slope=1, col="red" )

```

However, there *does* seem to be a difference when it comes to the consistency of F~st~ bewteen scaffolds; here illustrated with just (the SSR-bearing) scaffolds 4, 8, 22, 60 & 77.


```{r adj fst among SSR scafs, results='hide', echo=FALSE, }

filter_fsts <- filter( fsts, SCAF=="Scaffold4" | SCAF=="Scaffold8" | SCAF=="Scaffold22" | SCAF=="Scaffold60" | SCAF=="Scaffold77" ) %>% melt()

filter( filter_fsts,  variable %in% grep( "w_fst", levels(filter_fsts$variable), value=TRUE ) ) %>% 
    ggplot( aes( SCAF, value, group=variable ) ) + geom_point( aes( col=variable )) + 
      geom_line( aes( col=variable )) + scale_colour_discrete( name="comparison", 
        labels=gsub( "sum_([A-Z]{3,4})_([A-Z]{3,4})_[wm]_[fstvar]{3}_", "\\1 vs. \\2", 
                     levels(filter_fsts$variable))[seq(1, 45, 3)] ) +
          ylab( "mean adjusted Fst" ) + xlab( "Scaffold" ) + ggtitle( "Adjusted Fst" )

```

Note the increase in crossing over and the greater variablility in some of the comparisons, particularly those involving *BAM* and *BAVA*.

```{r fst among SSR scafs, results='hide', echo=FALSE, }

filter( filter_fsts,  variable %in% grep( "m_fst", levels(filter_fsts$variable), value=TRUE ) ) %>% 
    ggplot( aes( SCAF, value, group=variable ) ) + geom_point( aes( col=variable )) + 
      geom_line( aes( col=variable )) + scale_colour_discrete( name="comparison", 
        labels=gsub( "sum_([A-Z]{3,4})_([A-Z]{3,4})_[wm]_[fstvar]{3}_", "\\1 vs. \\2", 
                     levels(filter_fsts$variable))[seq(1, 45, 3)] ) +
          ylab( "mean Fst" ) + xlab( "Scaffold" ) + ggtitle( "'raw' Fst" )

```

Taking a slightly deeper look, lets see if we can get some idea of the variation *within* scaffolds -- I'll start with the SSR-bearing scaffolds for want of a more compelling place to start.

```{r dist. Fst values along scaffolds, echo=FALSE, message=FALSE, warning=FALSE, fig.width=10}

wholedataframes <- list( APA_BAM=APA_BAM, APA_BAVA=APA_BAVA, APA_COB=APA_COB, APA_IVI=APA_IVI, APA_MOV=APA_MOV, BAM_BAVA=BAM_BAVA, BAM_COB=BAM_COB, BAM_IVI=BAM_IVI, BAM_MOV=BAM_MOV, BAVA_COB=BAVA_COB, BAVA_IVI=BAVA_IVI, BAVA_MOV=BAVA_MOV, COB_IVI=COB_IVI, COB_MOV=COB_MOV, IVI_MOV=IVI_MOV )

for (i in 1:15) {   wholedataframes[[i]] <- mutate( wholedataframes[[i]], COMP=factor( rep( names(wholedataframes[i]), dim( wholedataframes[[i]] )[1] )), 
                                    BIN=paste( CHROM, "_", BIN_START, "_", BIN_END, coll="", sep="" ),
                                    ScafN=sub( "Scaffold", "", CHROM ) ) %>% 
                                  filter( WEIGHTED_FST > 0, MEAN_FST > 0 )  }

all_fsts <- wholedataframes$APA_BAM

for (i in 2:15) all_fsts <- rbind( all_fsts, wholedataframes[[i]] )


filter( all_fsts, CHROM=="Scaffold4" | CHROM=="Scaffold8" | CHROM=="Scaffold22" | CHROM=="Scaffold60" | CHROM=="Scaffold77" ) %>% 
      ggplot( aes( BIN_START, WEIGHTED_FST, col=COMP ) ) + geom_smooth( se=FALSE, method='loess' ) + facet_wrap( ~CHROM ) + 
              scale_colour_discrete( name="comparison", labels=sub( "_", " vs. ", levels( all_fsts$COMP ) )) + 
              xlab( "bin start coord." ) + ylab( "adjusted Fst" )

fst_var_scaf <- filter( all_fsts, CHROM=="Scaffold4" | CHROM=="Scaffold8" | CHROM=="Scaffold22" | CHROM=="Scaffold60" | CHROM=="Scaffold77" ) %>% 
                        group_by( COMP, CHROM )  %>% summarise( fst_var=var( WEIGHTED_FST ) )

FVSsum <- group_by( fst_var_scaf, COMP ) %>% summarise( MFV=mean( fst_var ) ) %>% arrange( desc( MFV ) )

```

So in this plot we can see a few things; firstly, that the ordering of the comparisons doesn't change an awful lot between *or* along these scaffolds. Secondly, there is noticeably more variability along scaffolds 60 & 77 than along the (longer) 4, 22 & 8.

To put that numerically the variance in F~st~ along these ranges from `r round(min(fst_var_scaf$fst_var),5)` to `r round(max(fst_var_scaf$fst_var),3)`. The `r FVSsum[1,1]` comparison is the most variable over all (mean var. of `r FVSsum[1,2]`), and `r FVSsum[15,1]` the least variable (mean var of `r FVSsum[15,2]`).

---

## Section 2

For a quick comparison, I'm going to throw together the same plots using the results files generated with a 200kb window, stepped at 100kb intervals.

```{r read in 2nd file set, results='hide', echo=FALSE, message=FALSE}

APA_BAM <- read.table( "Fsts/Fst_APA_vs_BAM.windowed200kb.step100kb.weir.fst", header=TRUE )
APA_BAVA <- read.table( "Fsts/Fst_APA_vs_BAVA.windowed200kb.step100kb.weir.fst", header=TRUE )
APA_COB <- read.table( "Fsts/Fst_APA_vs_COB.windowed200kb.step100kb.weir.fst", header=TRUE )
APA_IVI <- read.table( "Fsts/Fst_APA_vs_IVI.windowed200kb.step100kb.weir.fst", header=TRUE )
APA_MOV <- read.table( "Fsts/Fst_APA_vs_MOV.windowed200kb.step100kb.weir.fst", header=TRUE )
BAM_BAVA <- read.table( "Fsts/Fst_BAM_vs_BAVA.windowed200kb.step100kb.weir.fst", header=TRUE )
BAM_COB <- read.table( "Fsts/Fst_BAM_vs_COB.windowed200kb.step100kb.weir.fst", header=TRUE )
BAM_IVI <- read.table( "Fsts/Fst_BAM_vs_IVI.windowed200kb.step100kb.weir.fst", header=TRUE )
BAM_MOV <- read.table( "Fsts/Fst_BAM_vs_MOV.windowed200kb.step100kb.weir.fst", header=TRUE )
BAVA_COB <- read.table( "Fsts/Fst_BAVA_vs_COB.windowed200kb.step100kb.weir.fst", header=TRUE )
BAVA_IVI <- read.table( "Fsts/Fst_BAVA_vs_IVI.windowed200kb.step100kb.weir.fst", header=TRUE )
BAVA_MOV <- read.table( "Fsts/Fst_BAVA_vs_MOV.windowed200kb.step100kb.weir.fst", header=TRUE )
COB_IVI <- read.table( "Fsts/Fst_COB_vs_IVI.windowed200kb.step100kb.weir.fst", header=TRUE )
COB_MOV <- read.table( "Fsts/Fst_COB_vs_MOV.windowed200kb.step100kb.weir.fst", header=TRUE )
IVI_MOV <- read.table( "Fsts/Fst_IVI_vs_MOV.windowed200kb.step100kb.weir.fst", header=TRUE )

```

```{r data munging 2, results='hide', echo=FALSE, message=FALSE}

sum_APA_BAM <- group_by( APA_BAM, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_APA_BAVA <- group_by( APA_BAVA, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_APA_COB <- group_by( APA_COB, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_APA_IVI <- group_by( APA_IVI, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_APA_MOV <- group_by( APA_MOV, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_BAM_BAVA <- group_by( BAM_BAVA, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_BAM_COB <- group_by( BAM_COB, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_BAM_IVI <- group_by( BAM_IVI, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_BAM_MOV <- group_by( BAM_MOV, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_BAVA_COB <- group_by( BAVA_COB, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_BAVA_IVI <- group_by( BAVA_IVI, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_BAVA_MOV <- group_by( BAVA_MOV, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )
sum_COB_IVI <- group_by( COB_IVI, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_COB_MOV <- group_by( COB_MOV, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )

sum_IVI_MOV <- group_by( IVI_MOV, CHROM ) %>%
                filter( WEIGHTED_FST > 0, MEAN_FST > 0 ) %>%
                summarise( w_fst=mean( WEIGHTED_FST ), m_fst=mean( MEAN_FST ), m_var=mean( N_VARIANTS ) )


mydataframes <- list( sum_APA_BAM=sum_APA_BAM, sum_APA_BAVA=sum_APA_BAVA, sum_APA_COB=sum_APA_COB, sum_APA_IVI=sum_APA_IVI, sum_APA_MOV=sum_APA_MOV, sum_BAM_BAVA=sum_BAM_BAVA, sum_BAM_COB=sum_BAM_COB, sum_BAM_IVI=sum_BAM_IVI, sum_BAM_MOV=sum_BAM_MOV, sum_BAVA_COB=sum_BAVA_COB, sum_BAVA_IVI=sum_BAVA_IVI, sum_BAVA_MOV=sum_BAVA_MOV, sum_COB_IVI=sum_COB_IVI, sum_COB_MOV=sum_COB_MOV, sum_IVI_MOV=sum_IVI_MOV )

cor_fst <- rep(NA, 15)

for (i in 1:15) {
  cor_fst[i] <- cor( mydataframes[[i]][2], mydataframes[[i]][3] )
  new_names <- paste( names( mydataframes[i] ), names( mydataframes[[i]] ), sep='_', coll='' )
  names( mydataframes[[i]] ) <- c( "SCAF", new_names[ 2:4 ] )
}

mean( cor_fst )

fsts <- mydataframes$sum_APA_BAM

for (i in 2:15) {
fsts <- merge( fsts, mydataframes[[i]], by.x="SCAF", by.y="SCAF", all.x=TRUE )
  }

```

The first thing to note is that the R^2^ for the correlation between the two F~st~ measures is still `r round(mean( cor_fst ),2)`.

```{r adj fst among SSR scafs 2, results='hide', echo=FALSE, }

filter_fsts <- filter( fsts, SCAF=="Scaffold4" | SCAF=="Scaffold8" | SCAF=="Scaffold22" | SCAF=="Scaffold60" | SCAF=="Scaffold77" ) %>% melt()

filter( filter_fsts,  variable %in% grep( "w_fst", levels(filter_fsts$variable), value=TRUE ) ) %>% 
    ggplot( aes( SCAF, value, group=variable ) ) + geom_point( aes( col=variable )) + 
      geom_line( aes( col=variable )) + scale_colour_discrete( name="comparison", 
        labels=gsub( "sum_([A-Z]{3,4})_([A-Z]{3,4})_[wm]_[fstvar]{3}_", "\\1 vs. \\2", 
                     levels(filter_fsts$variable))[seq(1, 45, 3)] ) +
          ylab( "mean adjusted Fst" ) + xlab( "Scaffold" ) + ggtitle( "Adjusted Fst" )

```

Secondly, we can see that this plot of mean adjusted F~st~ is *very* similar to the one above...

```{r dist. Fst values along scaffolds 2, echo=FALSE, message=FALSE, warning=FALSE, fig.width=10}

wholedataframes <- list( APA_BAM=APA_BAM, APA_BAVA=APA_BAVA, APA_COB=APA_COB, APA_IVI=APA_IVI, APA_MOV=APA_MOV, BAM_BAVA=BAM_BAVA, BAM_COB=BAM_COB, BAM_IVI=BAM_IVI, BAM_MOV=BAM_MOV, BAVA_COB=BAVA_COB, BAVA_IVI=BAVA_IVI, BAVA_MOV=BAVA_MOV, COB_IVI=COB_IVI, COB_MOV=COB_MOV, IVI_MOV=IVI_MOV )

for (i in 1:15) {   wholedataframes[[i]] <- mutate( wholedataframes[[i]], COMP=factor( rep( names(wholedataframes[i]), dim( wholedataframes[[i]] )[1] )), 
                                    BIN=paste( CHROM, "_", BIN_START, "_", BIN_END, coll="", sep="" ),
                                    ScafN=sub( "Scaffold", "", CHROM ) ) %>% 
                                  filter( WEIGHTED_FST > 0, MEAN_FST > 0 )  }

all_fsts <- wholedataframes$APA_BAM

for (i in 2:15) all_fsts <- rbind( all_fsts, wholedataframes[[i]] )


filter( all_fsts, CHROM=="Scaffold4" | CHROM=="Scaffold8" | CHROM=="Scaffold22" | CHROM=="Scaffold60" | CHROM=="Scaffold77" ) %>% 
      ggplot( aes( BIN_START, WEIGHTED_FST, col=COMP ) ) + geom_smooth( se=FALSE, method='loess' ) + facet_wrap( ~CHROM ) + 
              scale_colour_discrete( name="comparison", labels=sub( "_", " vs. ", levels( all_fsts$COMP ) )) + 
              xlab( "bin start coord." ) + ylab( "adjusted Fst" )

fst_var_scaf <- filter( all_fsts, CHROM=="Scaffold4" | CHROM=="Scaffold8" | CHROM=="Scaffold22" | CHROM=="Scaffold60" | CHROM=="Scaffold77" ) %>% 
                        group_by( COMP, CHROM )  %>% summarise( fst_var=var( WEIGHTED_FST ) )

FVSsum <- group_by( fst_var_scaf, COMP ) %>% summarise( MFV=mean( fst_var ) ) %>% arrange( desc( MFV ) )

```

...and thirdly, the pattern of F~st~ along the SSR-bearing scaffolds also looks pretty much the same. So, do we actually see differences between the results from differing window sizes anywhere?

```{r window effect plot, message=FALSE, echo=FALSE, results='hide', fig.width=15, fig.height=20}
# can we get a distribution of Fsts across the genome with varying window size?

comparisons <- c( "APA_vs_BAM", "APA_vs_BAVA", "APA_vs_COB", "APA_vs_IVI", "APA_vs_MOV", "BAM_vs_BAVA", "BAM_vs_COB", "BAM_vs_IVI", "BAM_vs_MOV", "BAVA_vs_COB", "BAVA_vs_IVI", "BAVA_vs_MOV", "COB_vs_IVI", "COB_vs_MOV", "IVI_vs_MOV" )

windows <- c( ".windowed100kb.step100kb.weir.fst",
              ".windowed200kb.step50kb.weir.fst",
              ".windowed500kb.step100kb.weir.fst",
              ".windowed100kb.step50kb.weir.fst",
              ".windowed300kb.step50kb.weir.fst",
              ".windowed200kb.step100kb.weir.fst",
              ".windowed400kb.step50kb.weir.fst" )

these_fsts <- data.frame()

for (i in 1:15) {
  for ( j in 1:length( windows ) ) {
    filename <- paste( "Fsts/Fst_", comparisons[i], windows[j], coll='', sep='' )
    filesize <- as.numeric( strsplit( gsub( '[ ]+', " ", system( paste( "wc -l ", filename, coll='', sep='' ),
                                                                 intern=T )), " ")[[1]][2])
    readin <- read.table( filename, header=TRUE, nrows=filesize  )
    readin <- transmute( readin, scaf=factor( CHROM ), bin=as.numeric( BIN_START ), n_var=as.numeric( N_VARIANTS ), 
                         fst=as.numeric( WEIGHTED_FST ), comp=factor( rep( comparisons[i], nrow( readin ) ) ),
                         winstep=factor( rep( gsub( '[a-z]', '', windows[j] ), nrow( readin ) ) ),
                         scaf_bin=factor( paste( scaf, '_', bin, sep='', coll='' ) ), 
                         scaf_comp=factor( paste( scaf, '_', comp, sep='', coll='' )) )  %>%  filter( fst > 0 )
    these_fsts <- rbind( these_fsts, readin )
}}


summary( these_fsts )

filter( these_fsts, n_var > 50 ) %>%
      ggplot( aes( scaf, fst, col=winstep ) ) + scale_colour_discrete( name="window/step (kb)" ) + 
              geom_point( alpha=0.1 ) + facet_wrap( ~comp, ncol=3 )
              
```

What's interesting here is the amount of overplotting; it seems to make very little difference which combination of window/step sizes I use, the points are mostly just stacking on top of one another. This may not be the most helpfuln way to look at the data however. I'm going to focus just on a subset of (semi-randomly picked) comparisons and see if I can't come up with a better way of looking at these data...

```{r window effect alt, message=FALSE, echo=FALSE, results='hide', fig.height=12, fig.width=12}

cowplot::plot_grid( filter( these_fsts, comp == "APA_vs_BAM" ) %>% ggplot( aes( x=fst, col=winstep ) ) + 
                      geom_density( aes( fill=winstep ), alpha=0.1 ) + ggtitle( "APA_vs_BAM" ), 
                    filter( these_fsts, comp == "APA_vs_COB" ) %>% ggplot( aes( x=fst, col=winstep ) ) + 
                      geom_density( aes( fill=winstep ), alpha=0.1 ) + ggtitle( "APA_vs_COB" ), 
                    filter( these_fsts, comp == "IVI_vs_MOV" ) %>% ggplot( aes( x=fst, col=winstep ) ) + 
                      geom_density( aes( fill=winstep ), alpha=0.1 ) + ggtitle( "IVI_vs_MOV" ), 
                    filter( these_fsts, comp == "IVI_vs_MOV" ) %>% ggplot( aes( x=fst, col=winstep ) ) + 
                      geom_density( aes( fill=winstep ), alpha=0.1 ) + ggtitle( "IVI_vs_MOV" ), 
                    align="h", ncol=2 )

```

Here we can see that the size of the window/step doesn't change the overall shape of the F~st~ distribution, but we *can* see that the differences are largest at height of the peaks. The choice of window/step sizes therefore may be more influential than we'd ideally like.

```{r fst density panel, echo=FALSE, message=FALSE, results='hide', fig.height=12, fig.width=12}

ggplot( these_fsts, aes( x=fst, col=comp ) ) + geom_density( fill="grey" ) + 
        geom_vline( xintercept=quantile( these_fsts$fst, probs=0.95 ), lty=2 ) +
        geom_vline( xintercept=quantile( these_fsts$fst, probs=0.9 ), lty=1 ) + facet_wrap( ~comp )

# # pick a win/step combination at semi-random:
# pdf( "density_100kb_window_50kb_step.pdf", height = 10, width = 10 )
#   filter( these_fsts, winstep == ".100.50.." ) %>% 
#       ggplot( aes( x=fst, col=comp ) ) + geom_density( fill="grey" ) + 
#           geom_vline( xintercept=quantile( these_fsts$fst, probs=0.95 ), lty=2 ) +
#           geom_vline( xintercept=quantile( these_fsts$fst, probs=0.9 ), lty=1 ) + facet_wrap( ~comp ) + ggtitle( "100kb window, 50kb step" )
# dev.off()

```

...and here is the plot of the distributions of (all) F~st~ values by comparison. Bear in mind that each of these distributions represents a superposition of distributions as above.

# Narrowing the Search Space

After discussing these distributions, we agreed to narrow down the set of values for me to work on. First up, let's concentrate on decently long scaffolds. Below is a summary of the distribution of scaffold sizes as measured (crudely) in no. windows (bins):

```{r n_bins summary, echo=FALSE, message=FALSE}

group_by( these_fsts, scaf ) %>% filter( winstep == ".500.100.." ) %>% summarise( n_bins=n_distinct( scaf_bin ) ) %>% arrange( desc( n_bins )) %>% summary()

```

With the mean at ~5 and the 3rd quartile at 2 it's clear that the distribution is going to be super-skewed (which is not unexpected), but this plot shows just how skewed a set of lengths we are dealing with:-

```{r n_bins density, echo=FALSE, message=FALSE, results='hide'}

group_by( these_fsts, scaf ) %>% filter( winstep == ".500.100.." ) %>% summarise( n_bins=n_distinct( scaf_bin ) ) %>% arrange( desc( n_bins )) %>% ggplot( aes(x=n_bins) ) + geom_density() + xlab( "Scaffold length (no. 500kb windows)" )

```

Luckily for us there's a strong relationship between scaffold length and the no. variants (see below): perhaps unsurprisingly, the longer scaffolds tend to bear more SNPs etc. There's a pretty obvious inflection point in this relationship at ~30 windows
(which is ~3.4Mb here) after which it becomes almost linear. This seems suggestive to me, but being conservative to start with I'm going to trim those scaffolds with length of <2 windows (~600kb); those on the left of the red line on the plot.

```{r scaf length v no SNPs, echo=FALSE, message=FALSE, results='hide'}

group_by( these_fsts, scaf ) %>% filter( winstep == ".500.100.." ) %>% summarise( n_bins=n_distinct( scaf_bin ), n_var=mean( n_var ) ) %>% arrange( desc( n_bins )) %>% mutate( badscaf=factor( ifelse( (scaf=="Scaffold1910" | scaf=="Scaffold3315" | scaf=="Scaffold3294" | scaf=="Scaffold24"), "C", "G")) ) %>% ggplot( aes(x=n_bins, y=n_var, col=badscaf) ) + geom_point() + geom_smooth( col="blue" ) + scale_x_log10() + xlab( "Scaffold length (log10 no. 500kb windows)" ) + ylab( "mean no. variants" ) + geom_vline( xintercept=2.5, col="red", lty=2 ) + theme( legend.position="none" ) + scale_color_manual( values=c("red", "black") )

```

By BLASTing the [VecScreen](http://www.ncbi.nlm.nih.gov/tools/vecscreen/univec/) sequences against our reference (see [my logbook](./WRP_P_kings_logbook.md)) I found evidence for potential contamination on 4 scaffolds. I'm simply going to exclude these from analysis as a first approach. Of these 4, only 1 (Scaffold24) is long enough to make it past the "length must be >2 windows" cutoff (marked in red on plot).

Lastly, as discussed, I'm going to narrow down that data to only those F~st~ values that are greater than the 90th percentile for all F~st~s. After this threefold trimming, the data we are left with looks like this:

```{r filter fsts, results='hide', echo=FALSE, message=FALSE}
# most scafs have only 1 bin. 3rd quartile at 2!
by_scaf <- group_by( these_fsts, scaf ) %>% summarise( n_bins=n_distinct( scaf_bin ) ) %>% filter( n_bins >2 )

fltr_fsts <- filter( these_fsts, scaf %in% droplevels(by_scaf$scaf) ) %>%  # dumping shorter scafs
              filter( scaf != "Scaffold1910" & scaf != "Scaffold3315" & 
                        scaf != "Scaffold3294" & scaf != "Scaffold24" )  %>% # dumping all 4 scafs with contamination
              group_by( comp ) %>% filter( fst > quantile( fst, probs=0.9 ) ) # dumping low fst scafs *within* comp

ggplot( fltr_fsts, aes( x=fst, col=comp ) ) + geom_density( fill="grey" ) + xlim(0,1) + facet_wrap( ~comp )

```

# Does no. SNPs predict F~st~?

Using the trimmed F~st~ dataset, can we see a relationship betweent the number of variants within a window and the F~st~ across that window...?

```{r fst vs n_var, echo=FALSE, message=FALSE, results='hide'}

ggplot( fltr_fsts, aes( y=fst, x=n_var, col=comp ) ) + geom_point() + facet_wrap(~comp)

```

There doesn't seem to be much of a relationship there at all --- I confess that I'm not sure what to make of that yet.

## Section 3

```{r window effect plot 2, message=FALSE, echo=FALSE, results='hide', fig.width=15, fig.height=20}
# can we get a distribution of Fsts across the genome with varying window size?

comparisons <- c( "APA_vs_BAM", "APA_vs_BAVA", "APA_vs_COB", "APA_vs_IVI", "APA_vs_MOV", "BAM_vs_BAVA", "BAM_vs_COB", "BAM_vs_IVI", "BAM_vs_MOV", "BAVA_vs_COB", "BAVA_vs_IVI", "BAVA_vs_MOV", "COB_vs_IVI", "COB_vs_MOV", "IVI_vs_MOV" )

windows <- c( ".windowed100kb.step100kb.weir.fst",
              ".windowed200kb.step50kb.weir.fst",
              ".windowed500kb.step100kb.weir.fst",
              ".windowed100kb.step50kb.weir.fst",
              ".windowed300kb.step50kb.weir.fst",
              ".windowed200kb.step100kb.weir.fst",
              ".windowed400kb.step50kb.weir.fst", 
              ".windowed10kb.step10kb.weir.fst", 
              ".windowed10kb.step5kb.weir.fst", 
              ".windowed5kb.step5kb.weir.fst" )

these_fsts <- data.frame()

for (i in 1:15) {
  for ( j in 1:length( windows ) ) {
    filename <- paste( "Fsts/Fst_", comparisons[i], windows[j], coll='', sep='' )
    filesize <- as.numeric( strsplit( gsub( '[ ]+', " ", system( paste( "wc -l ", filename, coll='', sep='' ),
                                                                 intern=T )), " ")[[1]][2])
    readin <- read.table( filename, header=TRUE, nrows=filesize  )
    readin <- transmute( readin, scaf=factor( CHROM ), bin=as.numeric( BIN_START ), n_var=as.numeric( N_VARIANTS ), 
                         fst=as.numeric( WEIGHTED_FST ), comp=factor( rep( comparisons[i], nrow( readin ) ) ),
                         winstep=factor( rep( gsub( '[a-z]', '', windows[j] ), nrow( readin ) ) ),
                         scaf_bin=factor( paste( scaf, '_', bin, sep='', coll='' ) ), 
                         scaf_comp=factor( paste( scaf, '_', comp, sep='', coll='' )) )  %>%  filter( fst > 0 )
    these_fsts <- rbind( these_fsts, readin )
}}


summary( these_fsts )

# filter( these_fsts, n_var > 50 ) %>%
#       ggplot( aes( scaf, fst, col=winstep ) ) + scale_colour_discrete( name="window/step (kb)" ) + 
#               geom_point( alpha=0.1 ) + facet_wrap( ~comp, ncol=3 )
              
```

OK, so here I wanted to assess how much signal we're losing by using longer window sizes. Here I'm plotting the distributions of F~st~ values for each combination of window size and step size. Note that the sets of stats with smaller window sizes are more right-tail-heavy in *all* comparisons, more strongly so in some than others. On this basis I'mn going to semi-arbitraily use the 5kb and 10kb non-overlapping-window result sets for the next bit...

```{r window effect alt panel, message=FALSE, echo=FALSE, results='hide', fig.height=26, fig.width=12}

cowplot::plot_grid( filter( these_fsts, comp == "APA_vs_BAM" ) %>% ggplot( aes( x=fst, col=winstep ) ) + 
                      geom_density( aes( fill=winstep ), alpha=0.1 ) + ggtitle( "APA_vs_BAM" ), 
                    filter( these_fsts, comp == "APA_vs_BAVA" ) %>% ggplot( aes( x=fst, col=winstep ) ) + 
                      geom_density( aes( fill=winstep ), alpha=0.1 ) + ggtitle( "APA_vs_BAVA" ), 
                    filter( these_fsts, comp == "APA_vs_COB" ) %>% ggplot( aes( x=fst, col=winstep ) ) + 
                      geom_density( aes( fill=winstep ), alpha=0.1 ) + ggtitle( "APA_vs_COB" ), 
                    filter( these_fsts, comp == "APA_vs_IVI" ) %>% ggplot( aes( x=fst, col=winstep ) ) + 
                      geom_density( aes( fill=winstep ), alpha=0.1 ) + ggtitle( "APA_vs_IVI" ), 
                    filter( these_fsts, comp == "APA_vs_MOV" ) %>% ggplot( aes( x=fst, col=winstep ) ) + 
                      geom_density( aes( fill=winstep ), alpha=0.1 ) + ggtitle( "APA_vs_MOV" ), 
                    filter( these_fsts, comp == "BAM_vs_BAVA" ) %>% ggplot( aes( x=fst, col=winstep ) ) + 
                      geom_density( aes( fill=winstep ), alpha=0.1 ) + ggtitle( "BAM_vs_BAVA" ), 
                    filter( these_fsts, comp == "BAM_vs_COB" ) %>% ggplot( aes( x=fst, col=winstep ) ) + 
                      geom_density( aes( fill=winstep ), alpha=0.1 ) + ggtitle( "BAM_vs_COB" ), 
                    filter( these_fsts, comp == "BAM_vs_IVI" ) %>% ggplot( aes( x=fst, col=winstep ) ) + 
                      geom_density( aes( fill=winstep ), alpha=0.1 ) + ggtitle( "BAM_vs_IVI" ), 
                    filter( these_fsts, comp == "BAM_vs_MOV" ) %>% ggplot( aes( x=fst, col=winstep ) ) + 
                      geom_density( aes( fill=winstep ), alpha=0.1 ) + ggtitle( "BAM_vs_MOV" ), 
                    filter( these_fsts, comp == "BAVA_vs_COB" ) %>% ggplot( aes( x=fst, col=winstep ) ) + 
                      geom_density( aes( fill=winstep ), alpha=0.1 ) + ggtitle( "BAVA_vs_COB" ), 
                    filter( these_fsts, comp == "BAVA_vs_IVI" ) %>% ggplot( aes( x=fst, col=winstep ) ) + 
                      geom_density( aes( fill=winstep ), alpha=0.1 ) + ggtitle( "BAVA_vs_IVI" ), 
                    filter( these_fsts, comp == "BAVA_vs_MOV" ) %>% ggplot( aes( x=fst, col=winstep ) ) + 
                      geom_density( aes( fill=winstep ), alpha=0.1 ) + ggtitle( "BAVA_vs_MOV" ), 
                    filter( these_fsts, comp == "COB_vs_IVI" ) %>% ggplot( aes( x=fst, col=winstep ) ) + 
                      geom_density( aes( fill=winstep ), alpha=0.1 ) + ggtitle( "COB_vs_IVI" ), 
                    filter( these_fsts, comp == "COB_vs_MOV" ) %>% ggplot( aes( x=fst, col=winstep ) ) + 
                      geom_density( aes( fill=winstep ), alpha=0.1 ) + ggtitle( "COB_vs_MOV" ), 
                    filter( these_fsts, comp == "IVI_vs_MOV" ) %>% ggplot( aes( x=fst, col=winstep ) ) + 
                      geom_density( aes( fill=winstep ), alpha=0.1 ) + ggtitle( "IVI_vs_MOV" ), 
                    align="h", ncol=3 )

# ggplot( these_fsts, aes( x=fst, col=winstep ) ) + geom_density( aes( fill=winstep ), alpha=0.1 ) + facet_wrap( ~comp, ncol=3 )

```

So, taking those results from the 5kb and 10kb non-overlapping-windows, I'm going to filter again in a similar way to above: dropping scaffolds shorter than 2 bins, dropping the 4 scafflods with potential contamination, and keeping the highest 10% of F~st~ values within each comparison.

```{r filter fsts again, results='hide', echo=FALSE, message=FALSE}
# most scafs have only 1 bin. 3rd quartile at 2!
by_scaf <- group_by( these_fsts, scaf ) %>% summarise( n_bins=n_distinct( scaf_bin ) ) %>% filter( n_bins >2 )

# here I'm selecting just the non-overlapping 5kb windows
fltr_fsts_5_5 <- filter( these_fsts, scaf %in% droplevels(by_scaf$scaf) ) %>%  # dumping shorter scafs
                 filter( scaf != "Scaffold1910" & scaf != "Scaffold3315" & 
                        scaf != "Scaffold3294" & scaf != "Scaffold24" )  %>% # dumping all 4 scafs with contamination
                  filter( winstep ==".5.5.." ) %>%    # selecting just the short non-overlapping windows
                  group_by( comp ) %>% filter( fst > quantile( fst, probs=0.9 ) ) # dumping low fst scafs *within* comp.

# and here I'm selecting just the non-overlapping 10kb windows
fltr_fsts_10_10 <- filter( these_fsts, scaf %in% droplevels(by_scaf$scaf) ) %>%  # dumping shorter scafs
                   filter( scaf != "Scaffold1910" & scaf != "Scaffold3315" & 
                        scaf != "Scaffold3294" & scaf != "Scaffold24" )  %>% # dumping all 4 scafs with contamination
                    filter( winstep ==".10.10.." ) %>%    # selecting just the short non-overlapping windows
                  group_by( comp ) %>% filter( fst > quantile( fst, probs=0.9 ) ) # dumping low fst scafs *within* comp.

ggplot( fltr_fsts_5_5, aes( x=fst, col=comp ) ) + geom_density( fill="grey" ) + xlim(0,1) + facet_wrap( ~comp )

```

With that filtered list of F~st~s, I'm then going to look for bins that occur in *both* comparisons between fixed penetrating & fixed non-penetrating stalks (BAVA vs. MOV & BAVA vs. IVI). Then I ask which of those bins *do not* occur in the comparison bewteen the phenotypically similar IVI & MOV populations. Finally I narrow down which of *those* bins also have elevated F~st~s in the comparison between the phenotypically mixed populations (APA vs. BAM).

```{r estimating numbers of divergent regions 90quant, results='hide', echo=FALSE, message=FALSE}
# Degree of overlap: loci of separation by drift should be arbitrary – can we ID regions where Fst is high in multiple comparisons (pop level and/or individual)

# BAVA p0 absent vs. IVI p0 present, MOV p0 present
# Add APA vs. BAM
# check if NAN's make it through

BAVA_vs_MOV_5_5_90 <- fltr_fsts_5_5$scaf_bin[ fltr_fsts_5_5$comp == "BAVA_vs_MOV" ]  # absent vs. present
BAVA_vs_IVI_5_5_90 <- fltr_fsts_5_5$scaf_bin[ fltr_fsts_5_5$comp == "BAVA_vs_IVI" ]  # absent vs. present
IVI_vs_MOV_5_5_90  <- fltr_fsts_5_5$scaf_bin[ fltr_fsts_5_5$comp == "IVI_vs_MOV" ]   # present vs. present
APA_vs_BAM_5_5_90  <- fltr_fsts_5_5$scaf_bin[ fltr_fsts_5_5$comp == "APA_vs_BAM" ]   # mostly present vs. mostly absent

#### 5kb by 5kb regions
length( intersect( BAVA_vs_MOV_5_5_90, BAVA_vs_IVI_5_5_90 ) ) # 2036 5kb bins appear in both absent vs. present comparisons...
length( setdiff( intersect( BAVA_vs_MOV_5_5_90, BAVA_vs_IVI_5_5_90 ), IVI_vs_MOV_5_5_90 ) )  # 1622 5kb bins appear in both absent vs. present comparisons *but not* in present vs. present comparison
length( intersect( setdiff( intersect( BAVA_vs_MOV_5_5_90, BAVA_vs_IVI_5_5_90 ), IVI_vs_MOV_5_5_90 ), APA_vs_BAM_5_5_90) )  # 51 5kb bins...*and* still notable in the messier comparison

###################

BAVA_vs_MOV_10_10_90 <- fltr_fsts_10_10$scaf_bin[ fltr_fsts_10_10$comp == "BAVA_vs_MOV" ]  # absent vs. present
BAVA_vs_IVI_10_10_90 <- fltr_fsts_10_10$scaf_bin[ fltr_fsts_10_10$comp == "BAVA_vs_IVI" ]  # absent vs. present
IVI_vs_MOV_10_10_90  <- fltr_fsts_10_10$scaf_bin[ fltr_fsts_10_10$comp == "IVI_vs_MOV" ]   # present vs. present
APA_vs_BAM_10_10_90  <- fltr_fsts_10_10$scaf_bin[ fltr_fsts_10_10$comp == "APA_vs_BAM" ]   # mostly present vs. mostly absent

#### 10kb by 10kb regions
length( intersect( BAVA_vs_MOV_10_10_90, BAVA_vs_IVI_10_10_90 ) ) # 1327 10kb bins appear in both absent vs. present comparisons...
length( setdiff( intersect( BAVA_vs_MOV_10_10_90, BAVA_vs_IVI_10_10_90 ), IVI_vs_MOV_10_10_90 ) )  # 959 10kb bins appear in both absent vs. present comparisons *but not* in present vs. present comparison
length( intersect( setdiff( intersect( BAVA_vs_MOV_10_10_90, BAVA_vs_IVI_10_10_90 ), IVI_vs_MOV_10_10_90 ), APA_vs_BAM_10_10_90) )  # 33 5kb bins...*and* still notable in the messier comparison

```

That done, I redid the whole proceedure, keeping the top 5% of values, rather than the top 10%.

```{r estimating numbers of divergent regions 95quant, results='hide', echo=FALSE, message=FALSE}
# Degree of overlap: loci of separation by drift should be arbitrary – can we ID regions where Fst is high in multiple comparisons (pop level and/or individual)

fltr_fsts_5_5 <- filter( these_fsts, scaf %in% droplevels(by_scaf$scaf) ) %>%  # dumping shorter scafs
                 filter( scaf != "Scaffold1910" & scaf != "Scaffold3315" & 
                        scaf != "Scaffold3294" & scaf != "Scaffold24" )  %>% # dumping all 4 scafs with contamination
                  filter( winstep ==".5.5.." ) %>%    # selecting just the short non-overlapping windows
                  group_by( comp ) %>% filter( fst > quantile( fst, probs=0.95 ) ) # dumping low fst scafs *within* comp.

fltr_fsts_10_10 <- filter( these_fsts, scaf %in% droplevels(by_scaf$scaf) ) %>%  # dumping shorter scafs
                   filter( scaf != "Scaffold1910" & scaf != "Scaffold3315" & 
                        scaf != "Scaffold3294" & scaf != "Scaffold24" )  %>% # dumping all 4 scafs with contamination
                    filter( winstep ==".10.10.." ) %>%    # selecting just the short non-overlapping windows
                  group_by( comp ) %>% filter( fst > quantile( fst, probs=0.95 ) ) # dumping low fst scafs *within* comp.

# BAVA p0 absent vs. IVI p0 present, MOV p0 present

BAVA_vs_MOV_5_5_95 <- fltr_fsts_5_5$scaf_bin[ fltr_fsts_5_5$comp == "BAVA_vs_MOV" ]  # absent vs. present
BAVA_vs_IVI_5_5_95 <- fltr_fsts_5_5$scaf_bin[ fltr_fsts_5_5$comp == "BAVA_vs_IVI" ]  # absent vs. present
IVI_vs_MOV_5_5_95  <- fltr_fsts_5_5$scaf_bin[ fltr_fsts_5_5$comp == "IVI_vs_MOV" ]   # present vs. present
APA_vs_BAM_5_5_95  <- fltr_fsts_5_5$scaf_bin[ fltr_fsts_5_5$comp == "APA_vs_BAM" ]   # mostly present vs. mostly absent

#### 5kb by 5kb regions
length( intersect( BAVA_vs_MOV_5_5_95, BAVA_vs_IVI_5_5_95 ) ) # 901 5kb bins appear in both absent vs. present comparisons...
length( setdiff( intersect( BAVA_vs_MOV_5_5_95, BAVA_vs_IVI_5_5_95 ), IVI_vs_MOV_5_5_95 ) )  # 901 5kb bins appear in both absent vs. present comparisons *but not* in present vs. present comparison
length( intersect( setdiff( intersect( BAVA_vs_MOV_5_5_95, BAVA_vs_IVI_5_5_95 ), IVI_vs_MOV_5_5_95 ), APA_vs_BAM_5_5_95) )  # 51 5kb bins...*and* still notable in the messier comparison


###################

BAVA_vs_MOV_10_10_95 <- fltr_fsts_10_10$scaf_bin[ fltr_fsts_10_10$comp == "BAVA_vs_MOV" ]  # absent vs. present
BAVA_vs_IVI_10_10_95 <- fltr_fsts_10_10$scaf_bin[ fltr_fsts_10_10$comp == "BAVA_vs_IVI" ]  # absent vs. present
IVI_vs_MOV_10_10_95  <- fltr_fsts_10_10$scaf_bin[ fltr_fsts_10_10$comp == "IVI_vs_MOV" ]   # present vs. present
APA_vs_BAM_10_10_95  <- fltr_fsts_10_10$scaf_bin[ fltr_fsts_10_10$comp == "APA_vs_BAM" ]   # mostly present vs. mostly absent

#### 10kb by 10kb regions
length( intersect( BAVA_vs_MOV_10_10_95, BAVA_vs_IVI_10_10_95 ) ) # 559 10kb bins appear in both absent vs. present comparisons...
length( setdiff( intersect( BAVA_vs_MOV_10_10_95, BAVA_vs_IVI_10_10_95 ), IVI_vs_MOV_10_10_95 ) )  # 435 10kb bins appear in both absent vs. present comparisons *but not* in present vs. present comparison
length( intersect( setdiff( intersect( BAVA_vs_MOV_10_10_95, BAVA_vs_IVI_10_10_95 ), IVI_vs_MOV_10_10_95 ), APA_vs_BAM_10_10_95) ) # 33 10kb bins...*and* still notable in the messier comparison

```

### Comparisons using *local* (i.e. within-comparison) F~st~ cutoff

window size | step size | %age Fst cutoff | # windows in both P vs. NP | ...**& not** in P vs. P | ...**and** in P~dominant~ vs. NP~dominant~
------------|-----------|-----------------|----------------------------|-------------------------|--------------------------------------------
5kb | 5kb | 90  | `r length( intersect( BAVA_vs_MOV_5_5_90, BAVA_vs_IVI_5_5_90 ) )`     | `r length( setdiff( intersect( BAVA_vs_MOV_5_5_90, BAVA_vs_IVI_5_5_90 ), IVI_vs_MOV_5_5_90 ) )`  | `r length( intersect( setdiff( intersect( BAVA_vs_MOV_5_5_90, BAVA_vs_IVI_5_5_90 ), IVI_vs_MOV_5_5_90 ), APA_vs_BAM_5_5_90) )`
10kb  | 10kb  | 90  | `r length( intersect( BAVA_vs_MOV_10_10_90, BAVA_vs_IVI_10_10_90 ) )` | `r length( setdiff( intersect( BAVA_vs_MOV_10_10_90, BAVA_vs_IVI_10_10_90 ), IVI_vs_MOV_10_10_90 ) )`  | `r length( intersect( setdiff( intersect( BAVA_vs_MOV_10_10_90, BAVA_vs_IVI_10_10_90 ), IVI_vs_MOV_10_10_90 ), APA_vs_BAM_10_10_90) )`
5kb | 5kb | 95 | `r length( intersect( BAVA_vs_MOV_5_5_95, BAVA_vs_IVI_5_5_95 ) )`     | `r length( setdiff( intersect( BAVA_vs_MOV_5_5_95, BAVA_vs_IVI_5_5_95 ), IVI_vs_MOV_5_5_95 ) )`  | `r length( intersect( setdiff( intersect( BAVA_vs_MOV_5_5_95, BAVA_vs_IVI_5_5_95 ), IVI_vs_MOV_5_5_95 ), APA_vs_BAM_5_5_95) )`
10kb  | 10kb  | 95 | `r length( intersect( BAVA_vs_MOV_10_10_95, BAVA_vs_IVI_10_10_95 ) )` | `r length( setdiff( intersect( BAVA_vs_MOV_10_10_95, BAVA_vs_IVI_10_10_95 ), IVI_vs_MOV_10_10_95 ) )`  | `r length( intersect( setdiff( intersect( BAVA_vs_MOV_10_10_95, BAVA_vs_IVI_10_10_95 ), IVI_vs_MOV_10_10_95 ), APA_vs_BAM_10_10_95) )`

### ...Incorporating how many Scaffolds?

window size | step size | %age Fst cutoff | # windows in both P vs. NP | ...**& not** in P vs. P | ...**and** in P~dominant~ vs. NP~dominant~
------------|-----------|-----------------|----------------------------|-------------------------|--------------------------------------------
5kb         | 5kb       |  90             | `r nlevels(droplevels(fltr_fsts_5_5$scaf[fltr_fsts_5_5$scaf_bin %in% intersect( BAVA_vs_MOV_5_5_90, BAVA_vs_IVI_5_5_90 )]))` | `r nlevels(droplevels(fltr_fsts_5_5$scaf[fltr_fsts_5_5$scaf_bin %in% setdiff( intersect( BAVA_vs_MOV_5_5_90, BAVA_vs_IVI_5_5_90 ), IVI_vs_MOV_5_5_90 )]))` | `r nlevels(droplevels(fltr_fsts_5_5$scaf[fltr_fsts_5_5$scaf_bin %in% intersect( setdiff( intersect( BAVA_vs_MOV_5_5_90, BAVA_vs_IVI_5_5_90 ), IVI_vs_MOV_5_5_90 ), APA_vs_BAM_5_5_90)]))`
10kb        | 10kb      |  90             | `r nlevels(droplevels(fltr_fsts_10_10$scaf[fltr_fsts_10_10$scaf_bin %in% intersect( BAVA_vs_MOV_10_10_90, BAVA_vs_IVI_10_10_90 )]))` | `r nlevels(droplevels(fltr_fsts_10_10$scaf[fltr_fsts_10_10$scaf_bin %in% setdiff( intersect( BAVA_vs_MOV_10_10_90, BAVA_vs_IVI_10_10_90 ), IVI_vs_MOV_10_10_90 )]))` | `r nlevels(droplevels(fltr_fsts_10_10$scaf[fltr_fsts_10_10$scaf_bin %in% intersect( setdiff( intersect( BAVA_vs_MOV_10_10_90, BAVA_vs_IVI_10_10_90 ), IVI_vs_MOV_10_10_90 ), APA_vs_BAM_10_10_90)]))`
5kb         | 5kb       |  95             | `r nlevels(droplevels(fltr_fsts_5_5$scaf[fltr_fsts_5_5$scaf_bin %in% intersect( BAVA_vs_MOV_5_5_95, BAVA_vs_IVI_5_5_95 )]))` | `r nlevels(droplevels(fltr_fsts_5_5$scaf[fltr_fsts_5_5$scaf_bin %in% setdiff( intersect( BAVA_vs_MOV_5_5_95, BAVA_vs_IVI_5_5_95 ), IVI_vs_MOV_5_5_95 )]))` | `r nlevels(droplevels(fltr_fsts_5_5$scaf[fltr_fsts_5_5$scaf_bin %in% intersect( setdiff( intersect( BAVA_vs_MOV_5_5_95, BAVA_vs_IVI_5_5_95 ), IVI_vs_MOV_5_5_95 ), APA_vs_BAM_5_5_95)]))`
10kb        | 10kb      |  95             | `r nlevels(droplevels(fltr_fsts_10_10$scaf[fltr_fsts_10_10$scaf_bin %in% intersect( BAVA_vs_MOV_10_10_95, BAVA_vs_IVI_10_10_95 )]))` | `r nlevels(droplevels(fltr_fsts_10_10$scaf[fltr_fsts_10_10$scaf_bin %in% setdiff( intersect( BAVA_vs_MOV_10_10_95, BAVA_vs_IVI_10_10_95 ), IVI_vs_MOV_10_10_95 )]))` | `r nlevels(droplevels(fltr_fsts_10_10$scaf[fltr_fsts_10_10$scaf_bin %in% intersect( setdiff( intersect( BAVA_vs_MOV_10_10_95, BAVA_vs_IVI_10_10_95 ), IVI_vs_MOV_10_10_95 ), APA_vs_BAM_10_10_95)]))`

```{r no. div. regions w.global cutoff, echo=FALSE, message=FALSE, results='hide'}

by_scaf <- group_by( these_fsts, scaf ) %>% summarise( n_bins=n_distinct( scaf_bin ) ) %>% filter( n_bins >2 )

fltr_fsts_5_5_90 <- filter( these_fsts, scaf %in% droplevels(by_scaf$scaf) ) %>%  # dumping shorter scafs
                    filter( scaf != "Scaffold1910" & scaf != "Scaffold3315" & 
                        scaf != "Scaffold3294" & scaf != "Scaffold24" )  %>% # dumping all 4 scafs with contamination
                    filter( winstep ==".5.5.." ) %>% filter( fst > quantile( fst, probs=0.9 )) # selecting the short windows & dumping low fst scafs globally

fltr_fsts_10_10_90 <- filter( these_fsts, scaf %in% droplevels(by_scaf$scaf) ) %>%  # dumping shorter scafs
                      filter( scaf != "Scaffold1910" & scaf != "Scaffold3315" & 
                        scaf != "Scaffold3294" & scaf != "Scaffold24" )  %>% # dumping all 4 scafs with contamination
                      filter( winstep ==".10.10.." ) %>% filter( fst > quantile( fst, probs=0.9 )) # selecting the short windows & dumping low fst scafs globally

fltr_fsts_5_5_95 <- filter( these_fsts, scaf %in% droplevels(by_scaf$scaf) ) %>%  # dumping shorter scafs
                    filter( scaf != "Scaffold1910" & scaf != "Scaffold3315" & 
                        scaf != "Scaffold3294" & scaf != "Scaffold24" )  %>% # dumping all 4 scafs with contamination
                    filter( winstep ==".5.5.." ) %>% filter( fst > quantile( fst, probs=0.95 )) # selecting the short windows & dumping low fst scafs globally

fltr_fsts_10_10_95 <- filter( these_fsts, scaf %in% droplevels(by_scaf$scaf) ) %>%  # dumping shorter scafs
                      filter( scaf != "Scaffold1910" & scaf != "Scaffold3315" & 
                        scaf != "Scaffold3294" & scaf != "Scaffold24" )  %>% # dumping all 4 scafs with contamination
                      filter( winstep ==".10.10.." ) %>% filter( fst > quantile( fst, probs=0.95 )) # selecting the short windows & dumping low fst scafs globally

BAVA_vs_MOV_5_5_90 <- fltr_fsts_5_5_90$scaf_bin[ fltr_fsts_5_5_90$comp == "BAVA_vs_MOV" ]  # absent vs. present
BAVA_vs_IVI_5_5_90 <- fltr_fsts_5_5_90$scaf_bin[ fltr_fsts_5_5_90$comp == "BAVA_vs_IVI" ]  # absent vs. present
IVI_vs_MOV_5_5_90  <- fltr_fsts_5_5_90$scaf_bin[ fltr_fsts_5_5_90$comp == "IVI_vs_MOV" ]   # present vs. present
APA_vs_BAM_5_5_90  <- fltr_fsts_5_5_90$scaf_bin[ fltr_fsts_5_5_90$comp == "APA_vs_BAM" ]   # mostly present vs. mostly absent

BAVA_vs_MOV_10_10_90 <- fltr_fsts_10_10_90$scaf_bin[ fltr_fsts_10_10_90$comp == "BAVA_vs_MOV" ]  # absent vs. present
BAVA_vs_IVI_10_10_90 <- fltr_fsts_10_10_90$scaf_bin[ fltr_fsts_10_10_90$comp == "BAVA_vs_IVI" ]  # absent vs. present
IVI_vs_MOV_10_10_90  <- fltr_fsts_10_10_90$scaf_bin[ fltr_fsts_10_10_90$comp == "IVI_vs_MOV" ]   # present vs. present
APA_vs_BAM_10_10_90  <- fltr_fsts_10_10_90$scaf_bin[ fltr_fsts_10_10_90$comp == "APA_vs_BAM" ]   # mostly present vs. mostly absent

BAVA_vs_MOV_5_5_95 <- fltr_fsts_5_5_95$scaf_bin[ fltr_fsts_5_5_95$comp == "BAVA_vs_MOV" ]  # absent vs. present
BAVA_vs_IVI_5_5_95 <- fltr_fsts_5_5_95$scaf_bin[ fltr_fsts_5_5_95$comp == "BAVA_vs_IVI" ]  # absent vs. present
IVI_vs_MOV_5_5_95  <- fltr_fsts_5_5_95$scaf_bin[ fltr_fsts_5_5_95$comp == "IVI_vs_MOV" ]   # present vs. present
APA_vs_BAM_5_5_95  <- fltr_fsts_5_5_95$scaf_bin[ fltr_fsts_5_5_95$comp == "APA_vs_BAM" ]   # mostly present vs. mostly absent

BAVA_vs_MOV_10_10_95 <- fltr_fsts_10_10_95$scaf_bin[ fltr_fsts_10_10_95$comp == "BAVA_vs_MOV" ]  # absent vs. present
BAVA_vs_IVI_10_10_95 <- fltr_fsts_10_10_95$scaf_bin[ fltr_fsts_10_10_95$comp == "BAVA_vs_IVI" ]  # absent vs. present
IVI_vs_MOV_10_10_95  <- fltr_fsts_10_10_95$scaf_bin[ fltr_fsts_10_10_95$comp == "IVI_vs_MOV" ]   # present vs. present
APA_vs_BAM_10_10_95  <- fltr_fsts_10_10_95$scaf_bin[ fltr_fsts_10_10_95$comp == "APA_vs_BAM" ]   # mostly present vs. mostly absent

```

After discussion, I re-ran the analyses again, but this time I imposed the 90th (or 95th) percentile cutoff *globally* (i.e. basing the percentile on the pooled distribution of all F~st~s across all comparisons).

### Comparisons using *global* F~st~ cutoff

### Numbers of bins (windows)

window size | step size | %age Fst cutoff | # windows in both P vs. NP | ...**& not** in P vs. P | ...**and** in P~dominant~ vs. NP~dominant~
------------|-----------|-----------------|----------------------------|-------------------------|--------------------------------------------
5kb         | 5kb       |  90             | `r length( intersect( BAVA_vs_MOV_5_5_90, BAVA_vs_IVI_5_5_90 ) )`     | `r length( setdiff( intersect( BAVA_vs_MOV_5_5_90, BAVA_vs_IVI_5_5_90 ), IVI_vs_MOV_5_5_90 ) )`  | `r length( intersect( setdiff( intersect( BAVA_vs_MOV_5_5_90, BAVA_vs_IVI_5_5_90 ), IVI_vs_MOV_5_5_90 ), APA_vs_BAM_5_5_90) )`
10kb        | 10kb      |  90             | `r length( intersect( BAVA_vs_MOV_10_10_90, BAVA_vs_IVI_10_10_90 ) )` | `r length( setdiff( intersect( BAVA_vs_MOV_10_10_90, BAVA_vs_IVI_10_10_90 ), IVI_vs_MOV_10_10_90 ) )`  | `r length( intersect( setdiff( intersect( BAVA_vs_MOV_10_10_90, BAVA_vs_IVI_10_10_90 ), IVI_vs_MOV_10_10_90 ), APA_vs_BAM_10_10_90) )`
5kb         | 5kb       |  95             | `r length( intersect( BAVA_vs_MOV_5_5_95, BAVA_vs_IVI_5_5_95 ) )`     | `r length( setdiff( intersect( BAVA_vs_MOV_5_5_95, BAVA_vs_IVI_5_5_95 ), IVI_vs_MOV_5_5_95 ) )`  | `r length( intersect( setdiff( intersect( BAVA_vs_MOV_5_5_95, BAVA_vs_IVI_5_5_95 ), IVI_vs_MOV_5_5_95 ), APA_vs_BAM_5_5_95) )`
10kb        | 10kb      |  95             | `r length( intersect( BAVA_vs_MOV_10_10_95, BAVA_vs_IVI_10_10_95 ) )` | `r length( setdiff( intersect( BAVA_vs_MOV_10_10_95, BAVA_vs_IVI_10_10_95 ), IVI_vs_MOV_10_10_95 ) )`  | `r length( intersect( setdiff( intersect( BAVA_vs_MOV_10_10_95, BAVA_vs_IVI_10_10_95 ), IVI_vs_MOV_10_10_95 ), APA_vs_BAM_10_10_95) )`

### ...Incorporating how many Scaffolds?

window size | step size | %age Fst cutoff | # windows in both P vs. NP | ...**& not** in P vs. P | ...**and** in P~dominant~ vs. NP~dominant~
------------|-----------|-----------------|----------------------------|-------------------------|--------------------------------------------
5kb         | 5kb       |  90             | `r nlevels(droplevels(fltr_fsts_5_5_90$scaf[fltr_fsts_5_5_90$scaf_bin %in% intersect( BAVA_vs_MOV_5_5_90, BAVA_vs_IVI_5_5_90 )]))` | `r nlevels(droplevels(fltr_fsts_5_5_90$scaf[fltr_fsts_5_5_90$scaf_bin %in% setdiff( intersect( BAVA_vs_MOV_5_5_90, BAVA_vs_IVI_5_5_90 ), IVI_vs_MOV_5_5_90 )]))` | `r nlevels(droplevels(fltr_fsts_5_5_90$scaf[fltr_fsts_5_5_90$scaf_bin %in% intersect( setdiff( intersect( BAVA_vs_MOV_5_5_90, BAVA_vs_IVI_5_5_90 ), IVI_vs_MOV_5_5_90 ), APA_vs_BAM_5_5_90)]))`
10kb        | 10kb      |  90             | `r nlevels(droplevels(fltr_fsts_10_10_90$scaf[fltr_fsts_10_10_90$scaf_bin %in% intersect( BAVA_vs_MOV_10_10_90, BAVA_vs_IVI_10_10_90 )]))` | `r nlevels(droplevels(fltr_fsts_10_10_90$scaf[fltr_fsts_10_10_90$scaf_bin %in% setdiff( intersect( BAVA_vs_MOV_10_10_90, BAVA_vs_IVI_10_10_90 ), IVI_vs_MOV_10_10_90 )]))` | `r nlevels(droplevels(fltr_fsts_10_10_90$scaf[fltr_fsts_10_10_90$scaf_bin %in% intersect( setdiff( intersect( BAVA_vs_MOV_10_10_90, BAVA_vs_IVI_10_10_90 ), IVI_vs_MOV_10_10_90 ), APA_vs_BAM_10_10_90)]))`
5kb         | 5kb       |  95             | `r nlevels(droplevels(fltr_fsts_5_5_95$scaf[fltr_fsts_5_5_95$scaf_bin %in% intersect( BAVA_vs_MOV_5_5_95, BAVA_vs_IVI_5_5_95 )]))` | `r nlevels(droplevels(fltr_fsts_5_5_95$scaf[fltr_fsts_5_5_95$scaf_bin %in% setdiff( intersect( BAVA_vs_MOV_5_5_95, BAVA_vs_IVI_5_5_95 ), IVI_vs_MOV_5_5_95 )]))` | `r nlevels(droplevels(fltr_fsts_5_5_95$scaf[fltr_fsts_5_5_95$scaf_bin %in% intersect( setdiff( intersect( BAVA_vs_MOV_5_5_95, BAVA_vs_IVI_5_5_95 ), IVI_vs_MOV_5_5_95 ), APA_vs_BAM_5_5_95)]))`
10kb        | 10kb      |  95             | `r nlevels(droplevels(fltr_fsts_10_10_95$scaf[fltr_fsts_10_10_95$scaf_bin %in% intersect( BAVA_vs_MOV_10_10_95, BAVA_vs_IVI_10_10_95 )]))` | `r nlevels(droplevels(fltr_fsts_10_10_95$scaf[fltr_fsts_10_10_95$scaf_bin %in% setdiff( intersect( BAVA_vs_MOV_10_10_95, BAVA_vs_IVI_10_10_95 ), IVI_vs_MOV_10_10_95 )]))` | `r nlevels(droplevels(fltr_fsts_10_10_95$scaf[fltr_fsts_10_10_95$scaf_bin %in% intersect( setdiff( intersect( BAVA_vs_MOV_10_10_95, BAVA_vs_IVI_10_10_95 ), IVI_vs_MOV_10_10_95 ), APA_vs_BAM_10_10_95)]))`



```{r}

fsts_out <- droplevels( fltr_fsts_5_5_90[fltr_fsts_5_5_90$scaf_bin %in% 
                        intersect( setdiff( intersect( BAVA_vs_MOV_5_5_90, BAVA_vs_IVI_5_5_90 ),
                                            IVI_vs_MOV_5_5_90 ),
                                            APA_vs_BAM_5_5_90), ])

write.csv( fsts_out, "Fst_shortlist.csv", row.names=FALSE, quote=FALSE )

```

