---
title: "MakeGenotypeTable"
author: "Will Pitchers"
date: "27/02/2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

MakeGenotypeTable This script is to make the plots discussed with JG during our meeting on 24th Feb. 2017

The first set of input files called are those generated by `get_top_SNPs.qsub` running on the HPC.

```{r set up, include=FALSE}

rm( list=ls())

require( MASS )
require( dplyr )
require( data.table )
require( ReporteRs )

R.Version()

```


```{r, include=FALSE}

# my_dir <- "~/Dropbox (MSU Efish Lab)/WILL/genome_assembly/garrulous-turtle/assoc_output/"
# my_file_list <- dir( my_dir )

```



```{r, include=FALSE}

setwd( "~/Dropbox (MSU Efish Lab)/WILL/genome_assembly/garrulous-turtle/assoc_output/" )

filtered_output <- full_join( tbl_df( fread( "all_fish_version_5-1_HPC.filtered.snps_GQ15_geno50.assoc.fisher.csv" )),
                            tbl_df( fread( "all_fish_version_5-1_HPC.filtered.snps_GQ15_geno50_topSNPs.table" )), 
                            by=c( "Scaf" = "CHROM", "SNP" = "POS") )

names( filtered_output ) <- gsub( ".GT", "", names( filtered_output ) )

write.csv( filtered_output, "filtered_output_GQ15_g50.csv", quote=FALSE )

system( paste( "cat filtered_output_GQ15_g50.csv | tr '/'", "'_'", "> temp && mv temp filtered_output_GQ15_g50.csv" ) )

```


```{r, include=FALSE}

p0present <- c( "APA_6675", "APA_6676", "APA_6677", "APA_6678", "APA_6679", "APA_6680", "APA_6682", "APA_6683", "APA_6684", "APA_6685", "APA_6737", "BAM_6597", "COB_4018", "COB_4019", "IVI_3923", "IVI_4816", "IVI_4832", "IVI_4834", "IVI_4893", "IVI_4894", "IVI_4895", "IVI_4896", "IVI_4897", "IVI_4921", "IVI_4925", "MOV_6716", "MOV_6717", "MOV_6718", "MOV_6719", "MOV_6720", "MOV_6721", "MOV_6722", "MOV_6723", "MOV_6724", "MOV_6725" )

p0absent <- c( "APA_6681", "BAM_6494", "BAM_6496", "BAM_6497", "BAM_6498", "BAM_6499", "BAM_6500", "BAM_6501", "BAM_6502", "BAM_6598", "BAM_6599", "BAM_6602", "BAM_6603", "BAM_6604", "BAM_6605", "BAVA_6619", "BAVA_6620", "BAVA_6621", "BAVA_6622", "BAVA_6623", "BAVA_6624", "BAVA_6625", "BAVA_6626", "BAVA_6627", "COB_4004", "COB_4006", "COB_4027", "COB_4029" )

assoc <- fread( "filtered_output_g100.csv" )

assoc <- assoc %>% mutate( hom_ref=paste( A1, "_", A1, sep='' ), hom_alt=paste( A2, "_", A2, sep='' ), 
                          het_1=paste( A1, "_", A2, sep='' ), het_2=paste( A2, "_", A1, sep='' ) )

assoc_table <- assoc[ 1:10, 2:73 ]
assoc <- assoc[ 1:10, ]

assocFT <- FlexTable( assoc_table, header.cell.props=cellProperties( text.direction="tbrl" ) )
assocFT <- setFlexTableWidths( assocFT, widths=c( rep(1, 9), rep(0.5, 63) ) )

names( assoc_table )

for( i in 10:72) {
    assocFT[ assoc_table[, i] == assoc$hom_ref, i ] = cellProperties( background.color= "blue" )
    assocFT[ assoc_table[, i] == assoc$hom_alt, i ] = cellProperties( background.color= "red" )
    assocFT[ assoc_table[, i] == assoc$het_1, i ] = cellProperties( background.color= "pink" )
    assocFT[ assoc_table[, i] == assoc$het_2, i ] = cellProperties( background.color= "pink" )
}

for( i in p0present ) {
  assocFT[, i, to='header' ] = cellProperties( background.color= "green" )
}

for( i in p0absent ) {
  assocFT[, i, to='header'] = cellProperties(  background.color= "yellow" )
}

# knitr::kable( assoc_table )

```

```{r, results='asis', echo=FALSE}

datatable( assoc )

cat( as.html( assocFT ))

```


